src/operators/fold_window.rs

```diff
@@ -35,14 +37,14 @@ impl FoldWindowLogic {
 }
 
 impl WindowLogic<TdPyAny, TdPyAny, Option<TdPyAny>> for FoldWindowLogic {
-    fn exec(&mut self, next_value: Option<TdPyAny>) -> Option<TdPyAny> {
+    fn with_next(&mut self, next_value: Option<TdPyAny>) -> Option<TdPyAny> {
```

Comment:
why name this one `with_next` and the other `awake_with`?

Reply:
Mmmm. Maybe that's more misleading then helpful. I was trying to remind us that the contexts in which these are called by the framework are slightly different:

- `StatefulLogic::exec` could happen at any old random time (although probably just your scheduled time) and you might not even have new data (but also probably do)! This is a random awakening in the night and you need to be prepared to do nothing useful.

- `WindowLogic::exec` specifically only happens when the window has new data (or we know the data stream is permanently over). This is like the "inverse"? the "b-side"? of `Iterator::next` -- you will be called with the result.

Yeah, the types `Poll<Option<V>>` vs `Option<V>` tell this to you as well, but I felt like the name `exec` was too generic anyway.

Something like `awaken` feels very appropriate for the `StatefulLogic::exec` because of the timeout dimension, but it feels not appropriate for the `WindowLogic::exec` because time is irrelevant and the iterator part seems the most important. (I couldn't find any prior art for this sort of iterator inversion of control in the std lib, otherwise I would have stolen that name.)

Perhaps `StatefulLogic::awake_with_next` and `StatefulLogic::with_next` to show the inner part is related, but stateful logic has that other timeout component?

Reply:
`exec_with` and `awake_with`? This explanation is helpful-- not too fussed about it, and i think the "sometimes there's a thing with the next" is the part that gets sticky for me.

appreciate the walkthrough